datafusion/optimizer/src/push_down_projection.rs

```diff
@@ -228,7 +228,7 @@ impl OptimizerRule for PushDownProjection {
                 // Gather all columns needed for expressions in this Aggregate
                 let mut new_aggr_expr = vec![];
                 for e in agg.aggr_expr.iter() {
-                    let column = Column::from_name(e.display_name()?);
+                    let column = Column::from(e.display_name()?);
```

Comment:
This allows one to correctly re-construct a column whose qualifier has been crammed into the name by aliasing, which is what I'm doing in `replace_distinct_aggregate` in this PR. Not sure whether this makes sense in general though.